Test/Unit/BoltTestCase.php

```diff
@@ -3,13 +3,318 @@
 namespace Bolt\Boltpay\Test\Unit;
 
 use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionObject;
+use ReflectionProperty;
+use PHPUnit\Framework\Constraint\IsType;
 
-class BoltTestCase extends TestCase
-{
-    protected function skipTestInUnitTestsFlow()
+if (!defined('PHP_CURRENT_VERSION')) {
+    define('PHP_CURRENT_VERSION', (float)phpversion());
+}
+
+if (PHP_CURRENT_VERSION < 7.2) {
+    class BoltTestCase extends TestCase
+    {
+        protected function skipTestInUnitTestsFlow()
+        {
+            if (!class_exists('\Magento\TestFramework\Helper\Bootstrap')) {
+                $this->markTestSkipped('Skip integration test in unit test flow');
+            }
+        }
+        
+        protected function setUp()
+        {
+            $this->setUpInternal();
+        }
+        
+        protected function tearDown()
+        {
+            parent::tearDown();
+            $this->tearDownInternal();
+        }
+        
+        protected function setUpInternal()
+        {
+            
+        }
+        
+        protected function tearDownInternal()
+        {
+            
+        }
+    }
+} else {
+    class BoltTestCase extends TestCase
     {
-        if (!class_exists('\Magento\TestFramework\Helper\Bootstrap')) {
-            $this->markTestSkipped('Skip integration test in unit test flow');
+        protected function skipTestInUnitTestsFlow()
+        {
+            if (!class_exists('\Magento\TestFramework\Helper\Bootstrap')) {
+                $this->markTestSkipped('Skip integration test in unit test flow');
+            }
+        }
+        
+        protected function setUp(): void
+        {
+            $this->setUpInternal();
+        }
+        
+        protected function tearDown(): void
+        {
+            parent::tearDown();
+            $this->tearDownInternal();
+        }
+        
+        protected function setUpInternal()
+        {
+
+        }
+        
+        protected function tearDownInternal()
+        {
+
+        }
+        
+        protected function createPartialMock($originalClassName, array $methods): \PHPUnit\Framework\MockObject\MockObject
```

Comment:
In PHPUnit 9, createPartialMock would check if the method does exist before creating Mock object, but some methods like `getValue, setValue, getTitle` are not defined as explicit methods, so it causes warning and fail to create Mock object.

Reply:
So you assume that if `PHP_CURRENT_VERSION < 7.2` when have PHPUnit 6.
Can we for better readability have this in condition?
Like if `PHP_CURRENT_VERSION < 7.2 OR PHP_UNIT_VERSION < 9.1`

Reply:
I have changed the conditional statement as you suggest.